storage-node/src/services/webApi/controllers/filesApi.ts

```diff
@@ -34,35 +34,45 @@ export async function getFile(
   res: express.Response<unknown, AppConfig>,
   next: express.NextFunction
 ): Promise<void> {
+  const { id: dataObjectId } = req.params
   try {
-    await pinDataObjectIdToCache(req.params.id)
+    pinDataObjectIdToCache(dataObjectId)
+  } catch (err) {
+    res.status(404).send()
+    return
+  }
 
-    const dataObjectId = new BN(req.params.id)
+  try {
     const uploadsDir = res.locals.uploadsDir
-
-    const fullPath = path.resolve(uploadsDir, dataObjectId.toString())
-
+    const fullPath = path.resolve(uploadsDir, dataObjectId)
     const fileInfo = await getFileInfo(fullPath)
-    const fileStats = await fsPromises.stat(fullPath)
+
+    // verify ipfs has of file we are about to serve, just in-case it is corrupt
```

Comment:
@zeeshanakram3 similar question to https://github.com/Joystream/joystream/pull/5016/files#r1438660236

Reply:
I understand the rationale, but I think it would affect the performance at scale, since hashing is CPU-bound work with linear time complexity (`O(n)`), so it could severely impact the throughput of `GET` requests if a large number of concurrent requests are trying to get some object especially if the object size is very large.

Reply:
Ok, that is a good point.